base/server/src/main/java/com/netscape/cmscore/dbs/LDAPPagedSearch.java

```diff
@@ -0,0 +1,130 @@
+// --- BEGIN COPYRIGHT BLOCK ---
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; version 2 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// (C) 2023 Red Hat, Inc.
+// All rights reserved.
+// --- END COPYRIGHT BLOCK ---
+package com.netscape.cmscore.dbs;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.netscape.certsrv.base.EBaseException;
+import com.netscape.certsrv.dbs.DBPagedSearch;
+import com.netscape.certsrv.dbs.EDBException;
+import com.netscape.certsrv.dbs.EDBNotAvailException;
+import com.netscape.certsrv.dbs.IDBObj;
+import com.netscape.cmscore.apps.CMS;
+
+import netscape.ldap.LDAPConnection;
+import netscape.ldap.LDAPControl;
+import netscape.ldap.LDAPException;
+import netscape.ldap.LDAPSearchConstraints;
+import netscape.ldap.LDAPSearchResults;
+import netscape.ldap.LDAPSortKey;
+import netscape.ldap.LDAPv3;
+import netscape.ldap.controls.LDAPPagedResultsControl;
+import netscape.ldap.controls.LDAPSortControl;
+
+/**
+ * Perform consecutive paged search until entries are available.
+ *
+ * @author Marco Fargetta {@literal <mfargett@redhat.com>}
+ */
+public class LDAPPagedSearch<E extends IDBObj>  extends DBPagedSearch<E> {
+    public static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(LDAPPagedSearch.class);
+
+    private DBRegistry registry;
+    private LDAPConnection conn = null;
+    private String base = null;
+    private String filter = null;
+    private String[] attrs = null;
+    private String sortKey = null;
+    private LDAPSearchResults res = null;
+
+    public LDAPPagedSearch(DBRegistry registry, LDAPConnection conn, String base, String filter, String[] attrs,
+            String sortKey) throws EBaseException {
+        this.registry = registry;
+        this.base = base;
+        this.filter = filter;
+        this.attrs = attrs;
+        this.sortKey = sortKey;
+        try {
+            this.conn = (LDAPConnection) conn.clone();
+        } catch (Exception e) {
+            throw new EBaseException(CMS.getUserMessage("CMS_BASE_CONN_FAILED",
+                        e.toString()));
```

Comment:
Let's chain the original exception object so we can get the full stack trace.

Reply:
This is the same here https://github.com/dogtagpki/pki/blob/57548bfeea4d6e8aa88180d1aa5de0e92a06ad43/base/server/src/main/java/com/netscape/cmscore/dbs/LDAPVirtualList.java#L101.

I am fixing only in this class for this PR.